### PR TITLE
DOC-7453: Document New Server Configuration Setting- Mobile External Browser

### LINKED ISSUE #7453: Help Wanted: Document New Client Configuration Property
# Context

https://github.com/mattermost/mattermost-mobile/pull/8220 introduces a new client configuration property that determines whether the mobile app should use the default external browser to perform SSO Authentication instead of the internal one.

# Documentation Ask

Source file: configure/site-configuration-settings.rst

Within the **Customization** section, add the new configuration setting (and its associated search metadata located directly above a section title) as a new section on this page.

Position the new configuration setting immediately following the **App custom URL schemes** section, and format the details of the section modelled after the [report a problem](https://docs.mattermost.com/configure/site-configuration-settings.html#report-a-problem-link) configuration setting.

Include the following configuration setting details:
- Section title: Mobile external browser
- Available only for self-hosted deployments
- Description: Configures the mobile app to perform SSO Authentication using the external default browser.
- String input. 
- System Config path: N/A
- config.json = NativeAppSettings -> MobileExternalBrowser
- Environment variable = MM_NATIVEAPPSETTINGS_MOBILEEXTERNALBROWSER

# QA Validation Criteria

In a generated docs preview:
- Searching for the new configuration setting returns the correct details.
- Visiting the docs page directly returns details about the new configuration setting.

### PR DESCRIPTION
**Pull request by Rutam21**
### Summary

This PR adds a Mobile External Browser section in the Site Configuration Settings page with the necessary configuration setting options available to the users.

### Ticket Link
Fixes #7453 

- [x] This PR follows the Mattermost Contributing Guidelines.

### CODE REVIEW

**Comment by Rutam21** [comment]
@cwarnermm Please review and suggest changes, if any. Thanks.

File: source/configure/site-configuration-settings.rst [hunk 1]
```diff
@@ -354,6 +354,18 @@ When configured, after OAuth or SAML user authentication is complete, custom URL
 | - Via mmctl: ``mmctl config set NativeAppSettings.AppCustomURLSchemes "mmauth://" "mmauthbeta://"``                                            |
 +------------------------------------------------------------------------------------------------------------------------------------------------+
```
[the remark this answers is not in the document]
Comment (Rutam21) [reply source/configure/site-configuration-settings.rst:356]: @cwarnermm I have added the metadata section above as suggested by you. Please review.

File: source/configure/site-configuration-settings.rst [hunk 2]
```diff
@@ -354,6 +354,25 @@ When configured, after OAuth or SAML user authentication is complete, custom URL
...
 
+.. config:setting:: custom-mobileexternalbrowser
+  :displayname: Mobile external browser (Customization)
+  :systemconsole: N/A
+  :configjson: .NativeAppSettings.MobileExternalBrowser
```
Comment (cwarnermm) [inline source/configure/site-configuration-settings.rst:360]:
```suggestion
  :configjson: NativeAppSettings.MobileExternalBrowser
```

**Comment by Rutam21** [comment]
The commits got messed up with multiple file changes. I will open a new PR in a while.

**Pull request closed without merging**